internal/stage_magnet_receive.go

```diff
@@ -31,8 +31,14 @@ func testMagnetReceiveExtendedHandshake(stageHarness *test_case_harness.TestCase
 	if err = assertExitCode(result, 0); err != nil {
 		return err
 	}
+
+	expected := fmt.Sprintf("Peer ID: %x\n", params.ExpectedPeerID)
```

Comment:
handshake stage is split in 3 steps, adding this check to have consistency in output

Reply:
Nice. One pattern we've found useful when multiple checks are present is to add lines with ticks for each successful check: 

<img width="622" alt="Screenshot 2024-09-12 at 10 51 35" src="https://github.com/user-attachments/assets/97fb00f9-8052-43fc-a97f-f813dc58a766">

(That way even if a later check fails, there's some re-assurance that your output is okay until xyz, it's just the new stuff that isn't implemented)